metadata-ingestion/src/datahub/ingestion/source/sql/vertica.py

```diff
@@ -18,254 +28,1381 @@
+        location: Optional[str] = None
+        try:
+            # SQLAlchemy stubs are incomplete and missing this method.
+            # PR: https://github.com/dropbox/sqlalchemy-stubs/pull/223.
+            table_info: dict = inspector.get_oauth_comment(model, schema)  # type: ignore
+        except NotImplementedError:
+            return description, properties, location
+        except ProgrammingError as error:
+            logger.debug(
+                f"Encountered ProgrammingError. Retrying with quoted schema name for schema {schema} and oauth {model} %s",
+                error,
+            )
+            table_info: dict = inspector.get_oauth_comment(model, f'"{schema}"')  # type: ignore
+        description = table_info.get("text")
+        if isinstance(description, tuple):
+            # Handling for value type tuple which is coming for dialect 'db2+ibm_db'
+            description = table_info["text"][0]
+        # The "properties" field is a non-standard addition to SQLAlchemy's interface.
+        properties = table_info.get("properties", {})
+        return description, properties, location
+
+    def is_dataset_eligible_for_profiling(
+        self,
+        dataset_name: str,
+        sql_config: SQLAlchemyConfig,
+        inspector: Inspector,
+        profile_candidates: Optional[List[str]],
+    ) -> bool:
+        """Return whether table, view or projection is  eligible for profiling
+        Args:
+            dataset_name (str): dataset name
+            sql_config (SQLAlchemyConfig): configuration
+            inspector (Inspector): inspector obj from reflection
+            profile_candidates (Optional[List[str]])
+        Returns:
+            bool: returns bool
+        """
+        return (
+            (
+                sql_config.table_pattern.allowed(dataset_name)
+                and sql_config.profile_pattern.allowed(dataset_name)
+            )
+            and (
+                sql_config.projection_pattern.allowed(dataset_name)
+                and sql_config.profile_pattern.allowed(dataset_name)
+            )
+            and (
+                profile_candidates is None
+                or (
+                    profile_candidates is not None
+                    and dataset_name in profile_candidates
+                )
+            )
+        )
+
+    def loop_profiler_requests_for_projections(
+        self,
+        inspector: Inspector,
+        schema: str,
+        sql_config: SQLAlchemyConfig,
+    ) -> Iterable["GEProfilerRequest"]:
+        """Function is used for collecting profiling related information for every projections
+            inside an schema.
+
+        Args: schema: schema name
+
+        Yields:
+            [type]: GEProfilerRequest object
+        """
+        from datahub.ingestion.source.ge_data_profiler import GEProfilerRequest
+
+        tables_seen: Set[str] = set()
+        profile_candidates = None  # Default value if profile candidates not available.
+        if (
```

Comment:
use super here

Reply:
this is a new functions , it does not exist in sql_common.py thatswhy we werent able to use super . ALso if we called loop_profiler_request for projections also there are local variabled - tables_seen: Set[str] = set()
        profile_candidates = None - which we wont be able to fetch from original function . this was reason for not using super here .